### DISCUSSION TITLE
I want the env file for oci - program service

### DISCUSSION THREAD

**Discussion by agamdahiya** [Sunbird-coKreat/Community#82] [Q&A]
https://github.com/ocisunbird/program-service  -- this is the url of the code i am setting up in my local and i dont have the env file . and also had query regarding the kafka , that the kafka is to install on local or not , if not then let me know the steps to run the code without kafka.

Find the screenshot.
![image](https://github.com/Sunbird-coKreat/Community/assets/127106035/75dc016b-1d29-4876-a6e8-bb2500aea642)

**Comment by Aparna-karri591** [comment]
Tagging @vaivk369 @vinukumar-vs

**Reply by agamdahiya** [reply]
@vinukumar-vs  can you provide me the env file.

**Reply by agamdahiya** [reply]
@Aparna-karri591  I haven't received any comment from the people you have tagged . can you please help me out to provide me the solve asap.

**Reply by Aparna-karri591** [reply]
@shubhambansaltarento Pls check this

**Comment by vaivk369** [comment]
Hi @agamdahiya, 
Here is the env file for your reference.

`const envVariables = {
    baseURL: process.env.dock_base_url || 'https://dock.sunbirded.org',
    SUNBIRD_URL: process.env.sunbird_base_url || 'https://dev.sunbirded.org',
    SUNBIRD_PORTAL_API_AUTH_TOKEN: process.env.sunbird_api_auth_token || '',
    DOCK_CHANNEL: process.env.dock_channel || 'sunbird',
    port: process.env.sunbird_program_port || 6000,
    CACHE_TTL: process.env.dock_cache_ttl || 900,
    level: process.env.sunbird_service_log_level || 'info',
    LEARNING_SERVICE_URL: process.env.learning_service_url || 'https://dock.sunbirded.org/action/',
    LEARNER_SERVICE_URL: process.env.learner_service_url,
    CONTENT_SERVICE_URL: process.env.content_service_url || 'https://dock.sunbirded.org/action/',
    OPENSABER_SERVICE_URL: process.env.opensaber_service_url || 'https://dock.sunbirded.org/content/reg',
    SUNBIRD_KAFKA_HOST: process.env.sunbird_kafka_host,
    DOCK_KAFKA_HOST: process.env.dock_kafka_host,
    DOCK_REDIS_HOST: process.env.dock_redis_host,
    DOCK_REDIS_PORT: process.env.dock_redis_port || 6379,
    SUNBIRD_AUTO_CREATION_TOPIC: process.env.sunbird_auto_creation_topic,
    SUNBIRD_QUESTION_BULKUPLOAD_TOPIC: process.env.sunbird_question_bulkupload_topic,
    SUNBIRD_KAFKA_BULKUPLOAD_CONSUMER_GROUP_ID:process.env.sunbird_kafka_bulkupload_consumer_group_id,
    SUNBIRD_ASSESSMENT_SERVICE_BASE_URL : process.env.sunbird_assessment_service_base_url,
    CORE_INGRESS_GATEWAY_IP: process.env.CORE_INGRESS_GATEWAY_IP,
    config: {
        user: process.env.sunbird_program_db_user || "postgres",
        host: process.env.sunbird_program_db_host || "localhost",
        database: process.env.sunbird_program_db_name || 'sunbird_programs',
        password: process.env.sunbird_program_db_password || 'password',
        port: process.env.sunbird_program_db_port || 5432,
        dialect: process.env.sunbird_program_db_dialect || "postgres",
        logging: false,
        pool: {
            max: process.env.sunbird_program_db_pool ? Number(process.env.sunbird_program_db_pool) : 100
        }
    },
    telemetryConfig: {
        host: process.env.telemetry_service_host,
        endpoint: process.env.telemetry_service_endpoint,
        method: 'POST'
    },
    SUNBIRD_GOOGLE_SERVICE_ACCOUNT_CREDENTIAL: {
        client_email: process.env.sunbird_google_oauth_client_email,
        private_key: process.env.sunbird_google_oauth_private_key
    }
}
module.exports = envVariables;

`

Please note that the user, host, database, password for the config param has to be set up to the correct details of your postgres database.

Also, please comment out the following line from app.js if you dont have kafka set up locally.

  qumlConsumerService.qumlConsumer();

**Comment by mmanimegalai** [comment]
@agamdahiya Did the above steps help ? Pls let us know if you need any further support, thanks

**Reply by agamdahiya** [reply]
I have tried but it is not working , because some api required Kafka consumption.

**Reply by vinukumar-vs** [reply]
@agamdahiya Can you explain in detail what issue are you facing? With the above comments how the team can help you to resolve the issue. 

Please share your analysis & specify exactly where you stuck. So that we can help on specific issue. 

From the below comment we can't help to resolve the issue. 
> because some api required Kafka consumption.

@mmanimegalai @vaivk369
